shell/platform/darwin/macos/framework/Source/FlutterChannelKeyResponder.mm

```diff
@@ -134,6 +134,8 @@ - (void)handleEvent:(NSEvent*)event callback:(FlutterAsyncKeyCallback)callback {
     default: {
       NSAssert(false, @"Unexpected key event type (got %lu).", event.type);
       callback(false);
+      // This should not happen. Return to suppress clang-tidy warning on `type` being nil.
+      return;
```

Comment:
If you're really sure this shouldn't happen, there's also:
```
std::unreachable();
```

The behaviour is undefined in the spec but typically compilers will treat this as "This will never happen so I'll `abort()` in debug builds, but I can optimise this branch out entirely in release builds."

Reply:
So I think I will not use `unreachable` since our plan here is NOT to crash in release build. If this branch is optimized away, we will pass nil to dictionary and crash.

Reply:
Yes -- in the current implementation of clang, but since it's undefined behaviour in the spec, a future version might format your hard drive while playing god save the queen. Either way, unless we're 100% certain there's no way we'll end up here, playing it safe seems wise! :)